drivers/serial/uart_stm32.c

```diff
@@ -278,6 +257,112 @@ static void uart_stm32_lpuart_set_baud_rate(struct device *dev,
 }
 #endif
 
+static inline void uart_stm32_set_baud_rate(struct device *dev,
+					    u32_t clock_rate, u32_t baud_rate)
+{
+#if defined(CONFIG_SOC_SERIES_STM32L0X) || defined(CONFIG_SOC_SERIES_STM32L4X)
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+
+	if (IS_LPUART_INSTANCE(UartInstance)) {
+		uart_stm32_lpuart_set_baud_rate(dev, clock_rate, baud_rate);
+	} else {
+		uart_stm32_usart_set_baud_rate(dev, clock_rate, baud_rate);
+	}
+#else
+	uart_stm32_usart_set_baud_rate(dev, clock_rate, baud_rate);
+#endif
+}
+
+#ifdef CONFIG_UART_LINE_CTRL
+static int uart_stm32_line_ctrl_set(struct device *dev,
+				    u32_t ctrl, u32_t val)
+{
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+	struct uart_stm32_data *data = DEV_DATA(dev);
+	int result = 0;
+
+#define change_if_not_equal(_name, _val)		\
+	do {						\
+		if (data->_name == _val) {		\
+			goto done;			\
+		} else {				\
+			data->_name = _val;		\
+		}					\
+	} while (0)
+
+	LL_USART_Disable(UartInstance);
+
+	switch (ctrl) {
+	case LINE_CTRL_BAUD_RATE:
+		change_if_not_equal(baud_rate, val);
+		uart_stm32_set_baud_rate(dev, data->clock_rate,
+					 data->baud_rate);
+		break;
+	case LINE_CTRL_RTS:
+		change_if_not_equal(rts_cts, val);
+		if (data->rts_cts) {
+			LL_USART_SetHWFlowCtrl(UartInstance,
+					       LL_USART_HWCONTROL_RTS_CTS);
+		} else {
+			LL_USART_SetHWFlowCtrl(UartInstance,
+					       LL_USART_HWCONTROL_NONE);
+		}
+		break;
+	default:
+		result = -ENOTSUP;
+		break;
+	}
+
+done:
+	LL_USART_Enable(UartInstance);
+	return result;
+}
+
+static int uart_stm32_line_ctrl_get(struct device *dev,
+				    u32_t ctrl, u32_t *val)
+{
+	struct uart_stm32_data *data = DEV_DATA(dev);
+
+	switch (ctrl) {
+	case LINE_CTRL_BAUD_RATE:
+		*val = data->baud_rate;
+		return 0;
+	case LINE_CTRL_RTS:
+		*val = data->rts_cts;
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
```

Comment:
This line could not be accessed, code coverage tools will complain.
I propose to remove default and return -ENOSUP here.